ci-operator: update docs on provided env (#75)
openshift/ci-docs: content/en/docs/architecture/step-registry.md

|:---|:---|:---|
|`${OPENSHIFT_CI}`|Set to `"true"`, should be used to detect that a script is running in a `ci-operator` environment. | Always. |
|`${SHARED_DIR}`|Directory on the step's filesystem where files shared between steps can be read and written.|Always.|
|`${ARTIFACT_DIR}`|Directory on the step's filesystem where files should be placed to persist them in the job's artifacts.|Always.|
|`${CLUSTER_PROFILE_DIR}`|Directory on the step's filesystem where credentials and configuration from the cluster profile are stored.|When the test as defined in a `ci-operator` configuration file sets a `cluster_profile.`|
|`${KUBECONFIG}`| Path to `system:admin` credentials for the ephemeral OpenShift cluster under test.|After an ephemeral cluster has been installed.|
|`${KUBEADMIN_PASSWORD_FILE}`| Path to the kubeadmin password file.|After an ephemeral cluster has been installed.|
|`${RELEASE_IMAGE_INITIAL}`|Image pull specification for the initial release payload snapshot when the test began to run.|When the test imports or builds an initial release. See the [docs](docs/architecture/ci-operator/#describing-inclusion-in-an-openshift-release).|

> |`${RELEASE_IMAGE_LATEST}`|Image pull specification for the ephemeral release payload used to install the ephemeral OpenShift cluster.|When the test imports or builds an initial release. . See the [docs](docs/architecture/ci-operator/#describing-inclusion-in-an-openshift-release).|

Reviewer: ```suggestion
|`${RELEASE_IMAGE_LATEST}`|Image pull specification for the ephemeral release payload used to install the ephemeral OpenShift cluster.|When the test imports or builds a target ("latest") release. . See the [docs](docs/architecture/ci-operator/#describing-inclusion-in-an-openshift-release).|
```
